allen-cell-animated/simularium-viewer
# Feature/custom geo

Problem
=======
This is the first step in converting to file format version 3 geo handling
 resolves #149 

Solution
========
I removed all the geometryFile request and handling. Thanks to @toloudis for some good discussion on the data processing. And added the data handling to the versionHandler to make sure all the new fields are present. 

## Type of change
Please delete options that are not relevant.

* New feature (non-breaking change which adds functionality)
* Breaking change (fix or feature that would cause existing functionality to not work as expected)
* This change requires a documentation update
* This change requires updated or new tests

Change summary:
---------------
* I started working on loading local files, I move that to another branch, but that's some of the early changes on here 
* refactor the path that used to use geometryFile to use the trajectoryFileInfo
* Add tests and typing
* Add error handling for user feedback

Steps to Verify:
----------------
1.  The existing example should work, although if they have geo, they'll only show as spheres now. 
2. I have some test files, I'll try to post them here if they arent too big, otherwise I'll share them on the dev channel. 

Keyfiles (delete if not relevant):
-----------------------
1. [src/simularium/VisGeometry.ts](https://github.com/allen-cell-animated/simularium-viewer/compare/feature/custom-geo?expand=1#diff-3a1fefcb5547dbeb43717d5829e7051157f9afe5eb71a06358a1d2e111adeef1)
2. [src/simularium/versionHandlers.ts](https://github.com/allen-cell-animated/simularium-viewer/compare/feature/custom-geo?expand=1#diff-c11b63de9abee754703762fead84e81e6a209e68bf3df4fb695e7483aa8d2ee7)

Thanks for contributing!
